zcash/address.go

```diff
@@ -96,8 +97,9 @@ func KeyToViewingKey(key []byte) ([]byte, error) {
 	if len(key) != 32 || key[0]&0xf0 != 0 {
 		return nil, ErrInvalidKey
 	}
-	viewKey := make([]byte, 32)
-	prfAddr(viewKey, key, 1)
+	viewKey := make([]byte, 64)
+	prfAddr(viewKey, key, 0)
+	prfAddr(viewKey[32:], key, 1)
```

Comment:
Per sections 4.2 and 5.4.6 of the Zcash spec, the output of `prfAddr(_, key, 1)` needs to be clamped to a valid Curve25519 secret key in order to match the definition of `sk_enc`. I'm guessing this was unnecessary in `askToPKenc()` because `curve25519.ScalarBaseMult()` performs clamping internally.